analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/PublicMethodArgumentsShouldBeCheckedForNullBase.cs

```diff
@@ -67,4 +83,10 @@ private static IOperation NullDereferenceCandidate(IOperation operation)
         };
         return candidate?.UnwrapConversion();
     }
+
+    private static IOperation AssignmentTarget(IOperation operation) =>
+        // Missing operation types: DeconstructionAssignment, CoalesceAssignment, CompoundAssignment, ThrowExpression
```

Comment:
Do we have an issue that tracks all the missing operations? If not, create one and link any other issues that mention parts of it. Add a [checklist](https://simpledev.io/lesson/task-list-md-1/) with all the missing operations to that issue.

Reply:
https://github.com/SonarSource/sonar-dotnet/issues/7091
I did not add for compound assignment - as it doesn't make much sense here -, nor for `ThrowExpression`. This seems to work ok:
```cs
public void ThrowExpression(object o)
{
    _ = o ?? throw new NullReferenceException(nameof(o));
    o.ToString();
}
```

Reply:
Please link to #7039 in the issue as well. Please add all cases we discussed so far. Even if they are working already, we need to make sure we have all cases under test. "This seems to work fine." is not enough.